prov/gni/src/gnix_mr_notifier.c

```diff
@@ -55,69 +57,69 @@ notifier_verify_stuff(struct gnix_mr_notifier *mrn) {
 }
 
 int
-_gnix_notifier_init(struct gnix_mr_notifier *mrn)
+_gnix_notifier_init(void)
 {
-	if (mrn == NULL) {
-		GNIX_INFO(FI_LOG_MR, "mr notifier NULL\n");
-		return -FI_EINVAL;
-	}
-
-	mrn->fd = 0;
-	mrn->cntr = NULL;
-	fastlock_init(&mrn->lock);
+	global_mr_not.fd = -1;
+	global_mr_not.cntr = NULL;
+	fastlock_init(&global_mr_not.lock);
```

Comment:
I don't think it was necessary for the ref count to be set here.

This function is now used to initialize the library, not each fabric.  It would only be run once.